packages/lib/getFeatures/wfs/buildWfsFilter.ts

```diff
@@ -23,14 +30,28 @@ const queryPrefix = ({
 const wfsLike = (
   fieldName: string,
   input: string,
-  { featurePrefix, useRightHandWildcard }: WfsParameters
-) => `
-<ogc:PropertyIsLike wildCard="*" singleChar="." escapeChar="!">
+  { featurePrefix, useRightHandWildcard, likeFilterAttributes }: WfsParameters
+) => {
+  const mergedLikeFilterAttributes = merge(
+    {},
+    defaultLikeFilterAttributes,
+    likeFilterAttributes
+  )
+  return `
+<ogc:PropertyIsLike${Object.entries(mergedLikeFilterAttributes).reduce(
+    (acc, [key, value]) => `${acc} ${key}="${value}"`,
+    ''
+  )}>
 <ogc:PropertyName>${featurePrefix}:${fieldName}</ogc:PropertyName>
 <ogc:Literal>${input}${
-  typeof useRightHandWildcard === 'boolean' && !useRightHandWildcard ? '' : '*'
-}</ogc:Literal>
+    typeof useRightHandWildcard === 'boolean' && !useRightHandWildcard
+      ? ''
+      : Object.hasOwn(mergedLikeFilterAttributes, 'wildCard')
+      ? mergedLikeFilterAttributes.wildCard
+      : '*'
```

Comment:
This might break for some services.

When I implemented the `wfsSearch` for the Masterportal, I found that the key for the three properties may vary as well and thus wrote the following documentation in the Masterportal

```md
#### Portalconfig.menu.tool.wfsSearch.searchInstance.requestConfig.likeFilter

Values inside a filter for a WFS service can be compared with an `equal` or a `like`.
If the comparison should be with a `like` then the filter needs additional properties. These may vary in value and property definition.
For the documentation, it is assumed that the properties are called `wildCard`, `singleChar` and `escapeChar`; variations like e.g. `single` and `escape` are possible and need to be configured in line with the service. All key-value pairs are used in the request as given.
```

If I now have a service that defines the properties as `wildcard`, `escape` and `singleChar`, the default values are being added to the Filter as well.
Also, when I define wildcard differently (value and key), the wrong wildcard char would be used.

I think it is easiest, if either all properties are defined on the object (if present) or none so the fallback case can be used.

Reply:
> If I now have a service that defines the properties as wildcard, escape and singleChar, the default values are being added to the Filter as well.
I know and this should not be a problem. Additional attributes should be ignored by the service. You could also add an attribute which is called `killserver=true` and it won`t kill the server. ;)

> Also, when I define wildcard differently (value and key), the wrong wildcard char would be used.
That's right. The only way to fix this is to splitt the configuration like this:
```
{
wildCardKey: "wild",
wildCardChar: "*"
}
```
But I am not happy with this and we will loose some flexibility by this. Because now it is also possible to add any arbitrary attributes to the filter operator. Which might be handy for the admin of the service to send custom data within.
So I chose the same way the masterportal did: the righthand wildcard key must be called "wildcard" (see [here](https://bitbucket.org/geowerkstatt-hamburg/masterportal/src/42f370cdeec25afa4a2b0aa41bb07e2c6e1c529f/src/modules/wfsSearch/js/buildFilter.js#lines-79)) Maybe we just add this last sentence analogously to the documentation?

> I think it is easiest, if either all properties are defined on the object (if present) or none so the fallback case can be used.
Even when all properties are defined on the object, we don`t know which one is the wildcard replacement. E.g. think about `{joker="*", esc="!", single="*"}`. There is no chance to guess which one is for the wildcard. So I thing the only correct way is mentioned above.

Reply:
📞 ✔️